enumerable.rb

```diff
@@ -0,0 +1,141 @@
+# frozen_string_literal: true
+
+module Enumerable # rubocop:disable Metrics/ModuleLength
+  def my_each
+    return to_enum unless block_given?
+
+    length.times do |x|
+      yield(self[x])
+    end
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    length.times do |x|
+      yield(x, self[x])
+    end
+    self
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    item = []
+    length.times do |x|
+      item.push(self[x]) if yield(self[x])
+    end
+    item
+  end
+
+  def my_all(field = nil) # rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+    return true if !block_given? && field.nil?
+
+    value = true
+    length.times do |x|
+      if !field.nil?
+        if field.instance_of? Regexp
+          value = false unless self[x].match(field)
+        elsif field.respond_to?(:is_a?) && (field.is_a?(String) || field.is_a?(Integer))
+          value = false if field != self[x]
+        elsif field.respond_to?(:is_a?) && !(self[x].is_a? field)
+          value = false
+        end
+      elsif block_given?
+        value = false unless yield(self[x])
+      end
+    end
+    value
+  end
+
+  def my_any(field = nil) # rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+    return true if !block_given? && field.nil?
+
+    value = false
+    length.times do |x|
+      if !field.nil?
+        if field.instance_of? Regexp
+          value = true if self[x].match(field)
+        elsif field.respond_to?(:is_a?) && (field.is_a?(String) || field.is_a?(Integer))
+          value = true if field == self[x]
+        elsif field.respond_to?(:is_a?) && (self[x].is_a? field)
+          value = true
+        end
+      elsif block_given?
+        value = true if yield(self[x])
+      end
+    end
+    value
+  end
+
+  def my_none(field = nil) # rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+    return true if !block_given? && field.nil?
+
+    length.times do |x|
```

Comment:
- [ ] my_none? returns true if the block never returns true for all elements
- [ ] my_none? when no block or argument is given returns true only if none of the collection members is true
- [ ] my_none? when a class is passed as an argument returns true if none of the collection is a member of such class
- [ ] my_none? when a Regex is passed as an argument returns true only if none of the collection matches the Regex
- [ ] my_none? when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern